modules/core/ui/theme/src/Form/GinContentFormBase.php

```diff
@@ -0,0 +1,223 @@
+   *
+   * @var \Drupal\Core\Datetime\DateFormatterInterface
+   */
+  protected $dateFormatter;
+
+  /**
+   * Constructs a ContentEntityForm object.
+   *
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
+   *   The entity repository service.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle service.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   * @param \Drupal\Core\Datetime\DateFormatterInterface $date_formatter
+   *   The date formatter service.
+   */
+  public function __construct(EntityRepositoryInterface $entity_repository, EntityTypeBundleInfoInterface $entity_type_bundle_info, TimeInterface $time, DateFormatterInterface $date_formatter) {
+    parent::__construct($entity_repository, $entity_type_bundle_info, $time);
+    $this->dateFormatter = $date_formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.repository'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('datetime.time'),
+      $container->get('date.formatter'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getNewRevisionDefault() {
+    return TRUE;
+  }
+
+  /**
+   * Function that returns an array of tab definitions to add.
+   *
+   * @return array
+   *   Array of tab definitions keyed by tab id. Each definition should
+   *   provide a location, title and weight.
+   */
+  protected function getFieldGroups() {
+    return [
+      'default' => [
+        'location' => 'main',
+        'title' => 'Default',
+        'weight' => -50,
+      ],
+      'meta' => [
+        'location' => 'sidebar',
+        'title' => $this->t('Meta'),
+        'weight' => 0,
+      ],
+      'location' => [
+        'location' => 'main',
+        'title' => $this->t('Location'),
+        'weight' => 50,
+      ],
+      'file' => [
+        'location' => 'main',
+        'title' => $this->t('Files'),
+        'weight' => 150,
+      ],
+      'revision' => [
+        'location' => 'sidebar',
+        'title' => $this->t('Revision information'),
+        'weight' => 200,
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+
+    // Add process callback to alter form after GinContentFormHelper.
+    $form['#process'][] = '::processContentForm';
+
+    // Only alter the form display if farm_ui_theme.use_field_group is TRUE
+    // or if the form display is new and not saved.
+    /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $form_display */
+    $form_display = $form_state->get('form_display');
+    if (!$form_display || !$form_display->getThirdPartySetting('farm_ui_theme', 'use_field_group', $form_display->isNew())) {
```

Comment:
I just encountered a bug caused by this logic. I was experimenting with adding a new field to Equipment assets through the UI, which created a `core.entity_form_display.asset.equipment.default` config entity. After I did this the content forms were broken on Equipment assets.

Screenshot:

![Screenshot from 2024-01-17 12-13-39](https://github.com/farmOS/farmOS/assets/95381/6dd79be6-d713-4f63-a716-a03a29140fe9)

Reply:
Ah yeah. It's really a feature tbh, so as to not break any existing form displays someone might have configured ;-)

We should be able to alter the form-display form to include this third party setting boolean. I didn't do this for now considering that modifying these view/form displays is a more advanced thing and we can expect the `use_field_group` third party setting to be configured manually on the config if needed. If you disagree we could try modifying the form here.

Reply:
I see what you mean! That makes sense.

However, I think we should hide the blank sidebar that appears in those cases. That's what "feels broken".

Is that easy to do?

Reply:
Oh the blank sidebar, hmm. This might be tricky because we tell Gin to use the content editing form on our `entity.{type}.add_form` and `entity.{type}.edit_form` routes so it might always try to apply these styling/changes regardless of the form mode.

But I have an idea. We can unset some of the changes that gin makes to this page, specifically the sidebar. And in the process we can still keep the "Save" button in the top right which is still a nice improvement for a standard form. It is a little hacky but works well. I'll push a commit up with this:

```php
    // In GinContentFormBase::processContentForm
    // Remove the sidebar if the display is not using field groups.
    /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $form_display */
    $form_display = $form_state->get('form_display');
    if (!$form_display || !$form_display->getThirdPartySetting('farm_ui_theme', 'use_field_group', $form_display->isNew())) {
      if (($index = array_search('gin/sidebar', $form['#attached']['library'])) !== false) {
        unset($form['#attached']['library'][$index]);
      }
      unset($form['gin_sidebar']);
      unset($form['gin_sidebar_overlay']);
      unset($form['gin_actions']['gin_sidebar_toggle']);
    }
```

Reply:
Example:
![Screenshot from 2024-01-18 18-54-10](https://github.com/farmOS/farmOS/assets/3116995/e5bc26ef-feef-41a0-bc9c-41366a9b7395)

Reply:
Hmm I'm still seeing a blank sidebar with this change. But it did remove the button for opening and closing it.

![Screenshot from 2024-01-20 12-22-59](https://github.com/farmOS/farmOS/assets/95381/c51b403f-e21d-4bf0-ae76-65ca0cd74dd8)

Reply:
This looks like it's working better now - thanks @paul121!

Yea too bad about the "Save" button at the top, but anyone who has custom entity displays can use the `use_field_group` settings.